### PR TITLE
New Barcode search 

### LINKED ISSUE #326: Search for specific barcodes in the Google Books app
When viewing the status board on the google books app https://sul-gbooks-stage.stanford.edu/statuses/), the user should be able to search for a specific barcode.

Suggested UI: text box to enter barcode and search button.  

If no barcode found, show this as a message.  If barcode found, go to the barcode detail page

### PR DESCRIPTION
**Pull request by jermnelson**
## Why was this change made?
Fixes #326 

## Was the documentation (README, DevOpsDocs, wiki) updated?
n/a

## Does this change affect how this application integrates with other services?
No
If so, please confirm:
- ~~[ ] change was tested on stage    and/or~~
- ~~[ ] test added to sul-dlss/infrastructure-integration-test~~

### CODE REVIEW

File: app/controllers/statuses_controller.rb [hunk 1]
```diff
@@ -40,4 +40,15 @@ def retry_book
   end
   # rubocop:enable Metrics/MethodLength
   # rubocop:enable Metrics/AbcSize
+
+  def search_barcode
+    barcode = params.require(:barcode)
+    status = params.require(:status)
```
Comment (jcoyne) [inline app/controllers/statuses_controller.rb]: why do we need status here?  Can we just `redirect_back(fallback_location: root_path)` if it's not found?

File: app/views/statuses/_barcode_search.html.erb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<%= form_with url: search_barcode_statuses_path do %>
+  <div class="form-row">
+    <div class="col-6">
+      <%= hidden_field_tag(:status, @status.id) %>
```
Comment (jcoyne) [inline app/views/statuses/_barcode_search.html.erb]: Then this can be dropped.

File: app/views/statuses/_barcode_search.html.erb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+      <%= text_field_tag :barcode, "", class: 'form-control form-control-sm', placeholder: 'Enter barcode' %>
```
Comment (jcoyne) [inline app/views/statuses/_barcode_search.html.erb]: probably ought to have a label tag for this field per the accessibility standards.

File: app/views/statuses/show.html.erb [hunk 4]
```diff
@@ -1,7 +1,16 @@
...
 <div>
-  <p>The Google Books DB keeps track of the state of Google Books.</p>
 
+  <div class="row">
+    <div class="col-7">
+      <p>The Google Books DB keeps track of the state of Google Books.</p>
+    </div>
+    <div class="col">
+      <div class="float-right mb-2">
+        <%= render partial: 'barcode_search' %>
```
Comment (jcoyne) [inline app/views/statuses/show.html.erb]: the `partial:` isn't necessary here, not a big concern though.

File: app/controllers/statuses_controller.rb (hunk 1 above)
Comment (jermnelson) [reply app/controllers/statuses_controller.rb]: Done; just pushed change.

File: app/views/statuses/_barcode_search.html.erb (hunk 3 above)
Comment (jermnelson) [reply app/views/statuses/_barcode_search.html.erb]: Added label with latest commit.

File: app/views/statuses/show.html.erb (hunk 4 above)
Comment (jermnelson) [reply app/views/statuses/show.html.erb]: Remove.

File: app/views/statuses/_barcode_search.html.erb (hunk 2 above)
Comment (jermnelson) [reply app/views/statuses/_barcode_search.html.erb]: Removed

**Pull request merged**